test/unit/configuration_test.rb

```diff
@@ -1,7 +1,12 @@
 require File.expand_path('../../test_helper', __FILE__)
 require 'mocha/configuration'
+require 'mocha/ruby_version'
 
 class ConfigurationTest < Mocha::TestCase
+  def teardown
+    Mocha::Configuration.reset_configuration
+  end
+
```

Comment:
Minor: I'd prefer we use `Configuration.override` passing a block containing the assertion in each of the tests so we don't need to add this `#teardown` method.

Reply:
Oh, do you mind if I do it in a separate PR? I think it would add noise to what is going to be a pretty big PR already :)

Reply:
Hmm but if there's no teardown, doesn't the configuration chagne carry over into other tests? For example, at the end of `test_allow_temporarily_changes_config_when_given_block`, `stubbing_method_unnecessarily` would be `:warn` instead of the default of `:allow`, which might cause false positives/negatives depending on the order that tests were run. I feel it's safer to reset the configuration after each test?

I also think `override` doesn't quite work for `test_enabling_strict_keyword_argument_matching_works_in_ruby_2_7_and_above`, because I am testing the setter. What do you think?

Reply:
> Hmm but if there's no teardown, wouldn't the configuration changes carry over into other tests? For example, at the end of `test_allow_temporarily_changes_config_when_given_block`, `stubbing_method_unnecessarily` would be `:warn` instead of the default of `:allow`, which might cause false positives/negatives depending on the order that tests were run. I feel it's safer to reset the configuration after each test?

Sorry - I had forgotten how this all worked and I had missed the calls to e.g. `Mocha.configure { |c| c.stubbing_method_unnecessarily = :allow }` at the beginning of the existing tests. I'm happy for you to leave the `teardown` method in place. 👍 

> I also think `override` doesn't quite work for `test_enabling_strict_keyword_argument_matching_works_in_ruby_2_7_and_above`, because I am testing the setter. What do you think?

Yes, of course - that makes perfect sense now I read the code more carefully - sorry!